addons/mail/models/mail_thread.py

```diff
@@ -4151,6 +4151,7 @@ def _message_auto_subscribe(self, updated_values, followers_existing_policy='ski
         """
         if not self:
             return True
+        self = self.browse(set(self.ids)) # Filter duplicates to avoid triggering mail_followers_res_partner_res_model_id_uniq constraint
```

Comment:
i'm not a big fan of doing a browse to fetch records we already have.  I'd do a set(self) and manually create a new recordset through the iteration of the set. I'm not sure it's that much better than your solution though... What do you think ?